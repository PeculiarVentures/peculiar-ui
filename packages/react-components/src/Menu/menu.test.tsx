import React from 'react';
import {
  renderWithWrapper as render,
  fireEvent,
  screen,
  act,
} from '../test-utils';
import { Menu } from './index';

describe('<Menu />', () => {
  const options: Partial<React.ComponentProps<typeof Menu>>['options'] = [
    {
      label: 'item-1',
    },
    {
      label: 'item-2',
    },
    {
      label: 'item-3',
    },
  ];

  it('should render with default styles', () => {
    const { asFragment } = render(
      <Menu
        options={options}
      >
        <button type="button">Open</button>
      </Menu>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', async () => {
    const { baseElement } = render(
      <Menu
        options={options}
        popoverProps={{
          className: 'my-class-name',
        }}
      >
        <button type="button">Open</button>
      </Menu>,
    );

    fireEvent.click(screen.getByRole('button'));

    // Popper update() - https://github.com/popperjs/react-popper/issues/350
    await act(async () => await null);

    expect(baseElement).toMatchSnapshot();
  });
});
