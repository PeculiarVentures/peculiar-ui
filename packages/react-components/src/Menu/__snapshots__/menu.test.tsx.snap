// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Menu /> > should pass className 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
}

.emotion-2 {
  outline: 0;
  max-width: calc(100% - 32px);
  min-width: 16px;
  max-height: calc(100% - 32px);
  min-height: 16px;
  border-radius: 4px;
  box-shadow: var(--pv-shadow-light-low);
  background-color: var(--pv-color-white);
}

.emotion-2[data-popper-placement^="bottom"] {
  margin: var(--pv-size-base-3) 0px;
}

.emotion-2[data-popper-placement^="top"] {
  margin: var(--pv-size-base-3) 0px;
}

.emotion-2[data-popper-placement^="right"] {
  margin: 0px var(--pv-size-base-3);
}

.emotion-2[data-popper-placement^="left"] {
  margin: 0px var(--pv-size-base-3);
}

.emotion-3 {
  padding: var(--pv-size-base) 0px;
  outline: none;
  list-style: none;
  margin: 0;
}

.emotion-4 {
  padding: 0px var(--pv-size-base-2);
  font-family: inherit;
  outline: none;
  width: 100%;
  height: var(--pv-size-base-7);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 200ms;
  transition: background-color 200ms;
  background-color: transparent;
  border: none;
  box-sizing: border-box;
  gap: var(--pv-size-base-2);
  cursor: pointer;
  color: var(--pv-color-black);
}

.emotion-4:hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-4:focus-visible,
.emotion-4[data-focused="true"] {
  background-color: var(--pv-color-gray-4);
}

.emotion-4:active,
.emotion-4[aria-selected="true"] {
  background-color: var(--pv-color-gray-5);
}

.emotion-5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<body>
  <div>
    <button
      aria-expanded="true"
      aria-haspopup="menu"
      type="button"
    >
      Open
    </button>
  </div>
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="my-class-name emotion-2"
      data-popper-escaped="true"
      data-popper-placement="bottom-end"
      data-popper-reference-hidden="true"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; position: absolute; left: auto; top: 0px; right: 0px; bottom: auto; transform: translate(0px, 0px);"
      tabindex="-1"
    >
      <ul
        class="emotion-3"
        role="menu"
        tabindex="-1"
      >
        <li
          class="emotion-4"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="emotion-5"
          >
            item-1
          </span>
        </li>
        <li
          class="emotion-4"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="emotion-5"
          >
            item-2
          </span>
        </li>
        <li
          class="emotion-4"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="emotion-5"
          >
            item-3
          </span>
        </li>
      </ul>
    </div>
  </div>
</body>
`;

exports[`<Menu /> > should render with default styles 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    type="button"
  >
    Open
  </button>
</DocumentFragment>
`;
